pkg/apis/authorization/validation/validation.go

```diff
@@ -99,3 +102,74 @@ func ValidateLocalSubjectAccessReview(sar *authorizationapi.LocalSubjectAccessRe
 
 	return allErrs
 }
+
+func validateResourceAttributes(resourceAttributes *authorizationapi.ResourceAttributes, fldPath *field.Path) field.ErrorList {
+	if resourceAttributes == nil {
+		return nil
+	}
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, validateFieldSelectorAttributes(resourceAttributes.FieldSelector, fldPath.Child("fieldSelector"))...)
+	allErrs = append(allErrs, validateLabelSelectorAttributes(resourceAttributes.LabelSelector, fldPath.Child("fieldSelector"))...)
+
+	return allErrs
+}
+
+func validateFieldSelectorAttributes(selector *authorizationapi.FieldSelectorAttributes, fldPath *field.Path) field.ErrorList {
+	if selector == nil {
+		return nil
+	}
+	allErrs := field.ErrorList{}
+
+	if len(selector.RawSelector) > 0 && len(selector.Requirements) > 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("rawSelector"), selector.RawSelector, "may not specified at the same time as requirements"))
+	}
+	if len(selector.RawSelector) == 0 && len(selector.Requirements) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("requirements"), fmt.Sprintf("when %s is specified, requirements or rawSelector is required", fldPath)))
+	}
+
+	for i, requirement := range selector.Requirements {
+		allErrs = append(allErrs, validateFieldSelectorRequirement(requirement, fldPath.Child("requirements").Index(i))...)
+	}
+
+	return allErrs
+}
+
+func validateFieldSelectorRequirement(requirement authorizationapi.FieldSelectorRequirement, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	switch requirement.Operator {
+	case metav1.LabelSelectorOpIn, metav1.LabelSelectorOpNotIn:
+		if len(requirement.Values) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("values"), "must be specified when `operator` is 'In' or 'NotIn'"))
+		}
+	case metav1.LabelSelectorOpExists, metav1.LabelSelectorOpDoesNotExist:
+		if len(requirement.Values) > 0 {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("values"), "may not be specified when `operator` is 'Exists' or 'DoesNotExist'"))
+		}
+	default:
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("operator"), requirement.Operator, "not a valid selector operator"))
```

Comment:
This means an aggregated server that supports theoretical future selector operators delegating authz to an older kube-apiserver would hard-fail the delegated authz call if it passed `requirements` in the SAR, but would fallback to authorizing unscoped access if it passed `rawSelector` (since the kube-apiserver `rawSelector` parse would fail).

That's weirdly asymmetric with the kube-apiserver pass-through of selector info on verbs that kube-apiserver thinks don't support selectors in order to support aggregated API servers from the future.

I don't think it's safe to propagate an unknown requirement, but I'm also not sure about hard-failing the SAR call

Reply:
I've adjusted so it doesn't fail here and instead will remove the requirement from the list evaluated.  So we evaluate a broader permissions check.